subsys/net/l2/ethernet/ethernet.c

```diff
@@ -492,6 +492,8 @@ static int ethernet_send(struct net_if *iface, struct net_pkt *pkt)
 	} else if (IS_ENABLED(CONFIG_NET_IPV6) &&
```

Comment:
As part of the #11229 I introduced L2 protocol field that stores the IEEE 802.3 protocol type field that is found in Ethernet header, it is found in commit ac56085fc077298fdda6eb20f0a008375dcb9edf. We could use that here which would allow to avoid having this kind of gptp specific info. Of course the network driver would need to set this after receiving the packet, and corresponding protocol handler would need to set it in TX. As a short term solution doing it like this is ok too.

Reply:
Myes, I have reviewed yet your RFC, but adding more stuff to net_pkt again is maybe not what we want. Also I am really not a fan of drivers that has to do an "L2 job" there (parsing eth header). But let's see the details in that PR.

Reply:
If we introduce this "l2_proto" flag into net_pkt, we could remove separate setting for the protocol family as that could be deducted from the l2_proto field. But we can discuss that later in #11229